packages/transformers/js/core/src/env_replacer.rs

```diff
@@ -187,19 +191,48 @@ impl<'a> Fold for EnvReplacer<'a> {
         }
       }
 
-      decls.push(decl.clone().fold_with(self));
+      let mut decl = decl.clone();
+      decl.visit_mut_with(self);
+      decls.push(decl);
     }
 
-    VarDecl {
+    *node = VarDecl {
       span: node.span,
       kind: node.kind,
       decls,
       declare: node.declare,
-    }
+    };
   }
 }
 
 impl<'a> EnvReplacer<'a> {
+  /// If an expression matches `process.browser = ...` then the RHS is replaced with
+  /// `true` when `is_browser` is set to true.
+  ///
+  /// This doesn't make sense so it's deprecated and will not run anymore on canary builds or
```

Comment:
Where is the logic to omit this code on canary builds and future versions?

Reply:
I haven't gated it yet